### PR TITLE
2024 01 09 tauri UI setup

### PR DESCRIPTION
**Pull request by mattyg**
Basic scaffolding of tauri app ui

- replace scaffolded svelte app with scaffolded sveltekit app
- add tailwind to sveltekit app
- add flowbite to sveltekit app
- implement basic empty pages with sidebar navigation

### CODE REVIEW

File: tauri-app/src/app.d.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// See https://kit.svelte.dev/docs/types#app
+// for information about these interfaces
+declare global {
+	namespace App {
+		// interface Error {}
```
Comment (thedavidmeister) [inline tauri-app/src/app.d.ts:5]: commented code

File: tauri-app/package.json [hunk 2]
```diff
@@ -1,26 +1,46 @@
...
-    "svelte-check": "^3.4.6",
-    "tslib": "^2.6.0",
-    "typescript": "^5.0.2",
-    "vite": "^5.0.0",
-    "@tauri-apps/cli": "^1.5.8"
-  }
+	"name": "tauri-app",
+	"version": "0.0.1",
+	"private": true,
+	"scripts": {
```
Comment (thedavidmeister) [inline tauri-app/package.json:5]: why npm scripts?

**Review by thedavidmeister — CHANGES_REQUESTED** [review]
we're using tabs for spaces?

everywhere else it's 2 spaces

**Comment by mattyg** [comment]
> we're using tabs for spaces?
> 
> everywhere else it's 2 spaces

fixed & updated eslint config to 2 spaces

File: tauri-app/package.json (hunk 2 above)
Comment (mattyg) [reply tauri-app/package.json]: generated by scaffolding and kept for reference but yes its redundant and unnecessary. removed.

File: tauri-app/src/app.d.ts (hunk 1 above)
Comment (mattyg) [reply tauri-app/src/app.d.ts]: removed

File: tauri-app/package.json (hunk 2 above)
Comment (mattyg) [reply tauri-app/package.json]: reverted -- added npm scripts back in to ensure users are not running incorrect versions of node dependency commands by using `npx`

**Pull request merged**